src/codecs/dxt.rs

```diff
@@ -102,7 +103,7 @@ impl<R: Read> DxtDecoder<R> {
         self.variant.decoded_bytes_per_block() as u64 * u64::from(self.width_blocks)
     }
 
-    fn read_scanline(&mut self, buf: &mut [u8]) -> io::Result<usize> {
+    fn read_scanline(&mut self, buf: &mut [u8], tmp: &mut Vec<u8>) -> io::Result<usize> {
```

Comment:
I'd rather not do this sort of micro-optimization as part of a larger PR. The only way to know whether it helps performance is to benchmark it, and that works better as a standalone PR where we can compare and discuss only one change.

Reply:
The problem is that it makes the code a bit more complicated and I'm not sure has any performance benefit at all.

Reply:
I don't disagree that growth in complexity can eventually cause problems, but pushing against any incremental increase in complexity is a slippery slope argument.

This change is tiny, local, and can be easily undone. 

When you extrapolate any tiny change into an eventual snowball of complexity, it becomes an extreme argument demanding literally zero increase in complexity. This makes most code changes impossible, and prevents the codebase from escaping its existing local maximum. 

The goal of keeping complexity in check is very good, but it needs to be applied with a broader perspective than stonewalling very simple changes, not for their actual immediate cost, but for extrapolated eventual imagined cost.

Reply:
Anyway, I've dropped the commits that removed redundant reallocations.